lib/spark_api/models/subresource.rb

```diff
@@ -16,10 +16,10 @@ def find_by_id(id, parent_id, arguments={})
 
       def parse_date_start_and_end_times(attributes)
         # Transform the date strings
-        unless attributes['Date'].nil?
+        unless attributes['Date'].nil? && attributes['Date'].empty?
           date = Date.strptime attributes['Date'], '%m/%d/%Y'
           ['StartTime','EndTime'].each do |time|
-            next if attributes[time].nil?
+            next if attributes[time].nil? && attributes['Date'].empty?
```

Comment:
Oops, I just noticed that both of these should be an `||`, unless I'm lost this late at night. The attribute can't be both null and empty :laughing:

Reply:
```
next if attributes[time].nil? && attributes[time].empty?
```